v2/sender/marshaller.go

```diff
@@ -0,0 +1,80 @@
+package sender
```

Comment:
can we try to keep this at the root of the package?
if there are things we don't want to expose, we can move them in an `internal` package in a refactoring